tests/test_collections.py

```diff
@@ -123,3 +125,200 @@ async def test_search_collections_bbox(app_client, mock_list_product_types, mock
 
     assert r.status_code == 200
     assert ["S2_MSI_L1C", "S1_SAR_GRD"] == [col["id"] for col in r.json().get("collections", [])]
+
+
+async def test_collections_pagination_default_and_custom_limits(app_client, mock_list_product_types):
+    """
+    Test pagination behavior for collections with default and custom limits.
+
+    This test ensures that:
+    - The default limit is applied when no limit parameter is provided.
+    - Custom limits adjust the number of returned collections and pagination links.
+    """
+
+    collections = []
+    for i in range(12):
+        collections.append(
+            {
+                "_id": f"sample_collection_{i}",
+                "ID": "sample_collection",
+                "title": "Sample Collection",
+            }
+        )
+
+    mock_list_product_types.return_value = collections
+
+    # Default limit returns 10 collections and correct pagination links
+    r = await app_client.get("/collections")
+    assert r.status_code == 200
+    assert r.json()["numberReturned"] == 10  # limit
+    assert r.json()["numberMatched"] == len(collections)
+    cols = r.json()["collections"]
+    assert len(cols) == 10
+    links = r.json()["links"]
+    assert len(links) == 3
+    assert {"next", "root", "self"} == {link["rel"] for link in links}
+
+    # Custom limit parameter adjusts returned collections and changes pagination links
+    r = await app_client.get("/collections", params={"limit": 12})
+    assert r.status_code == 200
+    assert r.json()["numberReturned"] == 12  # limit
+    assert r.json()["numberMatched"] == len(collections)
+    cols = r.json()["collections"]
+    assert len(cols) == 12
```

Comment:
Please use variables each time limit is set in `params`

```suggestion
    limit = 12
    r = await app_client.get("/collections", params={"limit": limit})
    assert r.status_code == 200
    assert r.json()["numberReturned"] == limit
    assert r.json()["numberMatched"] == len(collections)
    cols = r.json()["collections"]
    assert len(cols) == limit
```